Contributing Guide Updates

jsquire opened the pull request:
# Summary

The focus of these changes is to make minor formatting and content updates to the contributing guide.  Included changes:

- Fixes for some awkward phrasing
- Expanding some abbreviations for readability
- Updates to the test matrix to reflect the current state
- Update "master" to "main" for branch references
- Make the pattern of using a stable hash for codegen explicit
- Minor typo fixes

# References and Resources

- [[BUG] Documentation Inconsistency in contributing.md (#29758)](https://github.com/Azure/azure-sdk-for-net/issues/29758)

jsquire committed:
Contributing Guide Updates

The focus of these changes is to make minor formatting and content updates to the contributing guide.  Included:

- Fix awkward phrasing
- Expand some abbreviations for readability
- Update the test matrix
- Make the pattern of using a stable hash for codegen explicit

heaths reviewed (approved):
A couple instances for a small change, but otherwise LGTM. Did you also check there were no bookmark links to section headers you changed in this markdown file? I don't recall any, but from context here can't tell easily.

jsquire committed:
Address feedback

- "sign off" => "sign-off"
- Fix broken anchor link
- "master" => "main"
- A couple of additional phrasing tweaks

jsquire commented:
> Did you also check there were no bookmark links to section headers you changed in this markdown file? I don't recall any, but from context here can't tell easily.

Thank you!   I forgot to check and there was one impacted.   Fixed.

jsquire commented:
/check-enforcer override

jsquire commented on CONTRIBUTING.md:
```diff
@@ -462,7 +462,7 @@ See [Data Plane Quick Start Tutorial](https://github.com/Azure/azure-sdk-for-net
 
 - Prior to issuing the pull request:
-  - All code must have completed any necessary legal signoff for being publicly viewable (Patent review, JSR review, etc.)
+  - All code must have completed any necessary legal sign off for being publicly viewable (Patent review, JSR review, etc.)
```
Thanks for including the reasoning; I learned me something today.

heaths replied:
Given my history, I see this often with "setup" (noun) vs. "set up" (verb). `/sign\s?off/` is similar. In most cases it's not an issue, but I figured may as well be right in more customer-facing documentation.
